app/models/document_type/body_field.rb

```diff
@@ -5,7 +5,37 @@ def id
     "body"
   end
 
-  def type
-    "govspeak"
+  def payload(edition)
+    {
+      details: {
+        body: GovspeakDocument.new(edition.contents[id], edition).payload_html,
+      },
+    }
+  end
+
+  def updater_params(_edition, params)
+    params
+      .require(:revision)
+      .permit(contents: [:body])
+      .to_h
+      .deep_symbolize_keys
```

Comment:
Yeah, I thought it might look odd. I found working with `params` directly to be quite frustrating. For example `#to_h` actually returns the string key version, and I was worried our downstream code might be expecting one or the other. I think this means you'd need a slightly different `expect` statement - having tried a few!

I've had a go at rewriting the method to try and make it less surprising, but also more deterministic in the type of data it returns. Could you take another look?

Reply:
`to_h` on an ActionController::Parameters returns an instance of HashWithIndifferentAccess so you can work with either variant. 

The way you've changed it looks an improvement though 👍

Reply:
Huh, that's weird - I may not have used the combination of `#to_h` and `match` when I was trying to get it to work. Oh well - I still like the determinism.